### ISSUE TITLE
User config snippets are not loaded in nvim terminal

### ISSUE THREAD

**Issue by Booplicate** [fish-shell/fish-shell#10374]
Setup: `windows 10`, `alacritty 0.13.1`, `nvim 0.9.1`, `lunarvim 9ee3b7b`, `fish 3.7.0 (cygwin 3.5.1-1.x86_64)`, `$TERM` is `xterm-256color`

So I have a nice setup like this:
```sh
set -gx XDG_DATA_HOME $HOME/.local/share
set -gx XDG_CONFIG_HOME $HOME/.config
set -gx XDG_CACHE_HOME $TEMP
```
It's required to make everything work together. Thus my fish configs are at (as expected):
- main config `~/.config/fish/config.fish`
- various aliases `~/.config/fish/conf.d/aliases.fish`

If I run `type st` to get my alias, I get the expected output:
```sh
st is a function with definition
# Defined in ~/.config/fish/conf.d/aliases.fish @ line 14
function st --description 'git status shortcut'
    git status $argv
end
```
and it works if I run it.

Then in vim config I have this:
```lua
vim.opt.shell = "fish"
vim.opt.shellcmdflag = "-c"
vim.opt.shellquote = ""
vim.opt.shellxquote = ""
```
which allows me to use `fish` in lvim. Everything works fine, **however none of the aliases from `~/.config/fish/conf.d/aliases.fish` are defined**! I can see changes  from `~/.config/fish/config.fish` being applied (like prompt style), but none from the other config.
Then there's this:
1. If I move my alises from `~/.config/fish/conf.d/aliases.fish` to `~/.config/fish/config.fish`, then they work.
2. I added an `echo "foo"` in `aliases.fish` to see if it's being executed, while I can see it in the normal terminal, nothing is shown when I run `fish` within lvim.

So my question is, would you have any ideas what may prevent files from `~/.config/fish/conf.d/` from loading? As I understand those should load prior to the other config. I understand that there's so much going on, it's hard to tell if the problem is in `fish` or it's something else, maybe it's just my configurations are incorrect, just hoping for some tips about what should I look for.

**Comment by Booplicate** [comment]
In normal terminal I can see both configs being loaded:
![1](https://github.com/fish-shell/fish-shell/assets/53382877/354c91be-76f5-4045-a63e-ba84583cf029)
In nvim terminal only one is being loaded:
![2](https://github.com/fish-shell/fish-shell/assets/53382877/001dfa0a-208b-4e49-b5a8-e01e70e7dcc9)
(notice how it loads twice, apparently it's [a feature](https://github.com/akinsho/toggleterm.nvim/wiki/Per-file-type-shell) in [`toggleterm.nvim`](https://github.com/akinsho/toggleterm.nvim))

**Comment by faho** [comment]
> So I have a nice setup like this:

Note: Setting the XDG variables in config.fish basically doesn't really work because it is too late.

If you want to really use them for anything, you'll have to set them *outside* of fish. Of course here you set them to their defaults, so it doesn't really matter.

In this case I suspect that nvim starts fish with them set to something else, the config snippets are read according to that directory, then your config.fish is read but at that point the snippets are already *done*.

The snippets are `$__fish_config_dir/conf.d/*.fish $__fish_sysconf_dir/conf.d/*.fish $__fish_vendor_confdirs/*.fish`, you can see what you get there with e.g.

```fish
set -l snippets $__fish_config_dir/conf.d/*.fish $__fish_sysconf_dir/conf.d/*.fish $__fish_vendor_confdirs/*.fish
echo $snippets
```

**Comment by Booplicate** [comment]
Thank you for your assistance!

> Note: Setting the XDG variables in config.fish basically doesn't really work because it is too late.

I only set them there for nvim since fish was using the desired location already. I now set them on the system level so it's consistent (just in case).

> the config snippets are read according to that directory, then your config.fish is read

Which surprised me since if snippets are read from another directory, why config is read from the one I want?

> The snippets are `$__fish_config_dir/conf.d/*.fish $__fish_sysconf_dir/conf.d/*.fish $__fish_vendor_confdirs/*.fish`, you can see what you get there with

That helped debugging the issue:
1. fish can't use windows/mixed paths (probably a msys2 and cygwin issue). For example `set -l snippet $__fish_config_dir/conf.d/*.fish` fails with code 124. So does `ls C:/Users/User/.config/fish/conf.d/*.fish` with message `fish: No matches for wildcard`. If I use a unix path, it works. Perhaps fish uses similar code to get and execute the snippets and thus can't find any.
2. So I tried to use unix paths. However the variables would get overwritten to mixed paths once I'm in nvim terminal. It's probably some extra scripts from nvim which I'm not aware of.

I added the following code to `config.fish`:
```fish
for var in XDG_DATA_HOME XDG_CONFIG_HOME XDG_CACHE_HOME
    set -gx $var $(cygpath -u $$var)
end
```
and it worked because fish is being executed multiple times from nvim before it opens the terminal. Somewhat hacky, I will research for a proper fix later (maybe run a startup script for fish from nvim).
